PatternPal/PatternPal.Extension/Commands/SubscribeEvents.cs

```diff
@@ -410,8 +433,44 @@ private static LogEventResponse PushLog(LogEventRequest request)
         {
             LogProviderService.LogProviderServiceClient client =
                 new LogProviderService.LogProviderServiceClient(GrpcHelper.Channel);
-
-            return client.LogEvent(request);
+            try
+            {
+                LogEventResponse ler = client.LogEvent(request);
+                ServerStatusChanged?.Invoke();
+                switch (ler.Status)
+                {
+                    case LogStatusCodes.LscSuccess:
+                        ServerStatus = ServerStatusCodes.Available;
+                        return ler;
+                    case LogStatusCodes.LscFailure:
```

Comment:
You could use `case`-fall through for all `ServerStatusCode.Error`'s to improve readability, i.e.:
```C#
case LogStatusCodes.LscRejected:
case LogStatusCodes.LscInvalidArguments:
// ...
    ServerStatus = ServerStatusCodes.Error;
    return ler;
```